src/mcp/client/stdio.py

```diff
@@ -159,4 +165,92 @@ async def stdin_writer():
     ):
         tg.start_soon(stdout_reader)
         tg.start_soon(stdin_writer)
-        yield read_stream, write_stream
+        try:
+            yield read_stream, write_stream
+        finally:
+            # Clean up process
+            try:
+                process.terminate()
+                if sys.platform == "win32":
+                    try:
+                        with anyio.fail_after(2.0):
+                            await process.wait()
+                    except TimeoutError:
+                        # Force kill if it doesn't terminate
+                        process.kill()
+            except Exception:
+                pass
+
+
+def _get_executable_command(command: str) -> str:
+    """
+    Get the correct executable command normalized for the current platform.
+
+    Args:
+        command: Base command (e.g., 'uvx', 'npx')
+
+    Returns:
+        str: Platform-appropriate command
+    """
+
+    try:
+        if sys.platform != "win32":
+            return command
+        else:
+            # For Windows, we need more sophisticated path resolution
+            # First check if command exists in PATH as-is
+            command_path = shutil.which(command)
+            if command_path:
+                return command_path
```

Comment:
I prefer the more modern
```suggestion
            if command_path := shutil.which(command):
                return command_path
```